COMPAT: matplotlib v0.15rc1 issues

install via: ``conda install matplotlib -c tacaswell`` (only linux-64 ATM)

this is with ``pandas-0.17rc1`` 

```
......F......E..FF...FE..F./home/jreback/miniconda/envs/pandas/lib/python2.7/site-packages/matplotlib/__init__.py:872: UserWarning: axes.color_cycle is deprecated and replaced with axes.prop_cycle; please use the latter.
  warnings.warn(self.msg_depr % (key, alt_key))
/home/jreback/miniconda/envs/pandas/lib/python2.7/site-packages/matplotlib/__init__.py:892: UserWarning: axes.color_cycle is deprecated and replaced with axes.prop_cycle; please use the latter.
  warnings.warn(self.msg_depr % (key, alt_key))
...E.../home/jreback/pandas/pandas/tools/plotting.py:3242: UserWarning: To output multiple subplots, the figure containing the passed axes is being cleared
  "is being cleared", UserWarning)
EF......ES.....E.......E.E.........../home/jreback/miniconda/envs/pandas/lib/python2.7/site-packages/matplotlib/artist.py:221: MatplotlibDeprecationWarning: This has been deprecated in mpl 1.5, please use the
axes property.  A removal date has not been set.
  warnings.warn(_get_axes_msg, mplDeprecation, stacklevel=1)
.........FE..E........F.E..E..........E.......................EE.
======================================================================
ERROR: test_bar_colors (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 1810, in test_bar_colors
    self._check_colors(ax.patches[::5], facecolors=default_colors[:5])
TypeError: 'NoneType' object has no attribute '__getitem__'

======================================================================
ERROR: test_boxplot_colors (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 2860, in test_boxplot_colors
    _check_colors(bp, default_colors[0], default_colors[0], default_colors[2])
TypeError: 'NoneType' object has no attribute '__getitem__'

======================================================================
ERROR: test_df_legend_labels (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 2478, in test_df_legend_labels
    ax = df.plot(legend=True, secondary_y='b')
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 3610, in __call__
    sort_columns=sort_columns, **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2495, in plot_frame
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2323, in _plot
    plot_obj.generate()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 944, in generate
    self._make_plot()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1596, in _make_plot
    ax = self._get_ax(i)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1280, in _get_ax
    ax = self._maybe_right_yaxis(ax, i)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 976, in _maybe_right_yaxis
    new_ax._get_lines.color_cycle = orig_ax._get_lines.color_cycle
AttributeError: '_process_plot_var_args' object has no attribute 'color_cycle'

======================================================================
ERROR: test_errorbar_plot (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 3129, in test_errorbar_plot
    df.plot(yerr=df_err)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 3610, in __call__
    sort_columns=sort_columns, **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2495, in plot_frame
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2323, in _plot
    plot_obj.generate()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 944, in generate
    self._make_plot()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1609, in _make_plot
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1623, in _plot
    lines = MPLPlot._plot(ax, x, y_values, style=style, **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1241, in _plot
    return ax.errorbar(x, y, **kwds)
  File "/home/jreback/miniconda/envs/pandas/lib/python2.7/site-packages/matplotlib/__init__.py", line 1806, in inner
    return func(ax, *args, **kwargs)
  File "/home/jreback/miniconda/envs/pandas/lib/python2.7/site-packages/matplotlib/axes/_axes.py", line 2964, in errorbar
    raise ValueError("yerr must be a scalar, the same "
ValueError: yerr must be a scalar, the same dimensions as y, or 2xN.

======================================================================
ERROR: test_hist_colors (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 2737, in test_hist_colors
    self._check_colors(ax.patches[::10], facecolors=default_colors[:5])
TypeError: 'NoneType' object has no attribute '__getitem__'

======================================================================
ERROR: test_kde_colors_and_styles_subplots (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 2799, in test_kde_colors_and_styles_subplots
    for ax, c in zip(axes, list(default_colors)):
TypeError: 'NoneType' object is not iterable

======================================================================
ERROR: test_line_colors_and_styles_subplots (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 2634, in test_line_colors_and_styles_subplots
    for ax, c in zip(axes, list(default_colors)):
TypeError: 'NoneType' object is not iterable

======================================================================
ERROR: test_line_lim (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 1768, in test_line_lim
    ax = df.plot(secondary_y=True)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 3610, in __call__
    sort_columns=sort_columns, **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2495, in plot_frame
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2323, in _plot
    plot_obj.generate()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 944, in generate
    self._make_plot()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1579, in _make_plot
    if self._is_ts_plot():
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1572, in _is_ts_plot
    return not self.x_compat and self.use_index and self._use_dynamic_x()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1576, in _use_dynamic_x
    return _use_dynamic_x(self._get_ax(0), self.data)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1280, in _get_ax
    ax = self._maybe_right_yaxis(ax, i)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 976, in _maybe_right_yaxis
    new_ax._get_lines.color_cycle = orig_ax._get_lines.color_cycle
AttributeError: '_process_plot_var_args' object has no attribute 'color_cycle'

======================================================================
ERROR: test_subplots_dup_columns (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 1668, in test_subplots_dup_columns
    axes = df.plot(subplots=True, secondary_y='a')
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 3610, in __call__
    sort_columns=sort_columns, **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2495, in plot_frame
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2323, in _plot
    plot_obj.generate()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 944, in generate
    self._make_plot()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1579, in _make_plot
    if self._is_ts_plot():
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1572, in _is_ts_plot
    return not self.x_compat and self.use_index and self._use_dynamic_x()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1576, in _use_dynamic_x
    return _use_dynamic_x(self._get_ax(0), self.data)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1276, in _get_ax
    ax = self._maybe_right_yaxis(ax, i)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 976, in _maybe_right_yaxis
    new_ax._get_lines.color_cycle = orig_ax._get_lines.color_cycle
AttributeError: '_process_plot_var_args' object has no attribute 'color_cycle'

======================================================================
ERROR: test_subplots_sharex_axes_existing_axes (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 1651, in test_subplots_sharex_axes_existing_axes
    df['C'].plot(ax=axes[0], secondary_y=True)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 3436, in __call__
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2526, in plot_series
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2323, in _plot
    plot_obj.generate()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 944, in generate
    self._make_plot()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1579, in _make_plot
    if self._is_ts_plot():
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1572, in _is_ts_plot
    return not self.x_compat and self.use_index and self._use_dynamic_x()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1576, in _use_dynamic_x
    return _use_dynamic_x(self._get_ax(0), self.data)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1280, in _get_ax
    ax = self._maybe_right_yaxis(ax, i)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 976, in _maybe_right_yaxis
    new_ax._get_lines.color_cycle = orig_ax._get_lines.color_cycle
AttributeError: '_process_plot_var_args' object has no attribute 'color_cycle'

======================================================================
ERROR: test_df_series_secondary_legend (pandas.tests.test_graphics.TestSeriesPlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 889, in test_df_series_secondary_legend
    s.plot(legend=True, secondary_y=True)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 3436, in __call__
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2526, in plot_series
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2323, in _plot
    plot_obj.generate()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 944, in generate
    self._make_plot()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1579, in _make_plot
    if self._is_ts_plot():
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1572, in _is_ts_plot
    return not self.x_compat and self.use_index and self._use_dynamic_x()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1576, in _use_dynamic_x
    return _use_dynamic_x(self._get_ax(0), self.data)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1280, in _get_ax
    ax = self._maybe_right_yaxis(ax, i)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 976, in _maybe_right_yaxis
    new_ax._get_lines.color_cycle = orig_ax._get_lines.color_cycle
AttributeError: '_process_plot_var_args' object has no attribute 'color_cycle'

======================================================================
ERROR: test_errorbar_plot (pandas.tests.test_graphics.TestSeriesPlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 1113, in test_errorbar_plot
    s.plot(yerr=s_err)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 3436, in __call__
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2526, in plot_series
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2323, in _plot
    plot_obj.generate()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 944, in generate
    self._make_plot()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1609, in _make_plot
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1623, in _plot
    lines = MPLPlot._plot(ax, x, y_values, style=style, **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1241, in _plot
    return ax.errorbar(x, y, **kwds)
  File "/home/jreback/miniconda/envs/pandas/lib/python2.7/site-packages/matplotlib/__init__.py", line 1806, in inner
    return func(ax, *args, **kwargs)
  File "/home/jreback/miniconda/envs/pandas/lib/python2.7/site-packages/matplotlib/axes/_axes.py", line 2964, in errorbar
    raise ValueError("yerr must be a scalar, the same "
ValueError: yerr must be a scalar, the same dimensions as y, or 2xN.

======================================================================
ERROR: test_hist_secondary_legend (pandas.tests.test_graphics.TestSeriesPlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 852, in test_hist_secondary_legend
    df['b'].plot.hist(ax=ax, legend=True, secondary_y=True)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 3524, in hist
    return self(kind='hist', bins=bins, **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 3436, in __call__
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2526, in plot_series
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2323, in _plot
    plot_obj.generate()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 944, in generate
    self._make_plot()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1948, in _make_plot
    ax = self._get_ax(i)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1280, in _get_ax
    ax = self._maybe_right_yaxis(ax, i)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 976, in _maybe_right_yaxis
    new_ax._get_lines.color_cycle = orig_ax._get_lines.color_cycle
AttributeError: '_process_plot_var_args' object has no attribute 'color_cycle'

======================================================================
ERROR: test_ts_area_lim (pandas.tests.test_graphics.TestSeriesPlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 582, in test_ts_area_lim
    ax = tz_ts.plot.area(stacked=False, secondary_y=True)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 3560, in area
    return self(kind='area', **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 3436, in __call__
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2526, in plot_series
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2323, in _plot
    plot_obj.generate()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 944, in generate
    self._make_plot()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1579, in _make_plot
    if self._is_ts_plot():
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1572, in _is_ts_plot
    return not self.x_compat and self.use_index and self._use_dynamic_x()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1576, in _use_dynamic_x
    return _use_dynamic_x(self._get_ax(0), self.data)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1280, in _get_ax
    ax = self._maybe_right_yaxis(ax, i)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 976, in _maybe_right_yaxis
    new_ax._get_lines.color_cycle = orig_ax._get_lines.color_cycle
AttributeError: '_process_plot_var_args' object has no attribute 'color_cycle'

======================================================================
ERROR: test_ts_line_lim (pandas.tests.test_graphics.TestSeriesPlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 551, in test_ts_line_lim
    ax = self.ts.plot(secondary_y=True)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 3436, in __call__
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2526, in plot_series
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2323, in _plot
    plot_obj.generate()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 944, in generate
    self._make_plot()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1579, in _make_plot
    if self._is_ts_plot():
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1572, in _is_ts_plot
    return not self.x_compat and self.use_index and self._use_dynamic_x()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1576, in _use_dynamic_x
    return _use_dynamic_x(self._get_ax(0), self.data)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1280, in _get_ax
    ax = self._maybe_right_yaxis(ax, i)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 976, in _maybe_right_yaxis
    new_ax._get_lines.color_cycle = orig_ax._get_lines.color_cycle
AttributeError: '_process_plot_var_args' object has no attribute 'color_cycle'

======================================================================
FAIL: test_area_colors (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 2701, in test_area_colors
    self._check_colors(handles, linecolors=custom_colors)
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 194, in _check_colors
    self.assertEqual(len(collections), len(linecolors))
AssertionError: 10 != 5

======================================================================
FAIL: test_bar_log_no_subplots (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 2188, in test_bar_log_no_subplots
    tm.assert_numpy_array_equal(ax.yaxis.get_ticklocs(), expected)
  File "/home/jreback/pandas/pandas/util/testing.py", line 822, in assert_numpy_array_equal
    left.shape, right.shape)
  File "/home/jreback/pandas/pandas/util/testing.py", line 791, in raise_assert_detail
    raise AssertionError(msg)
AssertionError: numpy array are different

numpy array shapes are different
[left]:  (4,)
[right]: (2,)

======================================================================
FAIL: test_bar_log_subplots (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 2199, in test_bar_log_subplots
    tm.assert_numpy_array_equal(ax[0].yaxis.get_ticklocs(), expected)
  File "/home/jreback/pandas/pandas/util/testing.py", line 822, in assert_numpy_array_equal
    left.shape, right.shape)
  File "/home/jreback/pandas/pandas/util/testing.py", line 791, in raise_assert_detail
    raise AssertionError(msg)
AssertionError: numpy array are different

numpy array shapes are different
[left]:  (6,)
[right]: (4,)

======================================================================
FAIL: test_boxplot (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 2214, in test_boxplot
    self.bp_n_objects * len(numeric_cols))
AssertionError: 21 != 24

======================================================================
FAIL: test_boxplot_vertical (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 2251, in test_boxplot_vertical
    self.assertEqual(len(ax.lines), self.bp_n_objects * len(numeric_cols))
AssertionError: 21 != 24

======================================================================
FAIL: test_errorbar_scatter (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 3251, in test_errorbar_scatter
    _check_errorbar_color(ax.containers, 'green', has_err='has_yerr')
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 3240, in _check_errorbar_color
    self._check_colors(errs, linecolors=[expected] * len(errs))
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 204, in _check_colors
    self.assertEqual(result, expected)
AssertionError: Tuples differ: (0.0, 0.0, 1.0, 1.0) != (0.0, 0.5019607843137255, 0.0,...

First differing element 1:
0.0
0.501960784314

- (0.0, 0.0, 1.0, 1.0)
+ (0.0, 0.5019607843137255, 0.0, 1.0)

======================================================================
FAIL: test_subplots (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 1440, in test_subplots
    self._check_legend_labels(ax, labels=[com.pprint_thing(column)])
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 117, in _check_legend_labels
    self._check_text_labels(ax.get_legend().get_texts(), labels)
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 241, in _check_text_labels
    self.assertEqual(len(labels), len(expected))
AssertionError: 2 != 1

======================================================================
FAIL: test_bar_log (pandas.tests.test_graphics.TestSeriesPlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 649, in test_bar_log
    tm.assert_numpy_array_equal(ax.yaxis.get_ticklocs(), expected)
  File "/home/jreback/pandas/pandas/util/testing.py", line 822, in assert_numpy_array_equal
    left.shape, right.shape)
  File "/home/jreback/pandas/pandas/util/testing.py", line 791, in raise_assert_detail
    raise AssertionError(msg)
AssertionError: numpy array are different

numpy array shapes are different
[left]:  (6,)
[right]: (4,)

----------------------------------------------------------------------
Ran 136 tests in 124.794s

FAILED (SKIP=1, errors=15, failures=8)
```

cc @TomAugspurger @sinhrks @jorisvandenbossche 
cc @tacaswell

--- a/ci/install_conda.sh
+++ b/ci/install_conda.sh
@@ -72,6 +72,7 @@ bash miniconda.sh -b -p $HOME/miniconda || exit 1
 
 conda config --set always_yes yes --set changeps1 no || exit 1
 conda update -q conda || exit 1
+conda config --add channels conda-forge || exit 1
 conda config --add channels http://conda.binstar.org/pandas || exit 1
 conda config --set ssl_verify false || exit 1
 

--- a/ci/requirements-3.4.run
+++ b/ci/requirements-3.4.run
@@ -11,7 +11,6 @@ beautiful-soup
 scipy
 numexpr
 pytables
-matplotlib=1.3.1
 lxml
 sqlalchemy
 bottleneck

--- a/doc/source/whatsnew/v0.17.0.txt
+++ b/doc/source/whatsnew/v0.17.0.txt
@@ -50,6 +50,7 @@ Highlights include:
 - Documentation comparing SAS to *pandas*, see :ref:`here <compare_with_sas>`
 - Removal of the automatic TimeSeries broadcasting, deprecated since 0.8.0, see :ref:`here <whatsnew_0170.prior_deprecations>`
 - Compatibility with Python 3.5 (:issue:`11097`)
+- Compatibility with matplotlib 1.5.0 (:issue:`11111`)
 
 Check the :ref:`API Changes <whatsnew_0170.api>` and :ref:`deprecations <whatsnew_0170.deprecations>` before updating.
 

--- a/pandas/tests/test_graphics.py
+++ b/pandas/tests/test_graphics.py
@@ -2062,6 +2062,19 @@ class TestDataFramePlots(TestPlotBase):
         float_array = np.array([0.0, 1.0])
         df.plot.scatter(x='A', y='B', c=float_array, cmap='spring')
 
+    def test_scatter_colors(self):
+        df = DataFrame({'a': [1, 2, 3], 'b': [1, 2, 3], 'c': [1, 2, 3]})
+        with tm.assertRaises(TypeError):
+            df.plot.scatter(x='a', y='b', c='c', color='green')
+
+        ax = df.plot.scatter(x='a', y='b', c='c')
+        tm.assert_numpy_array_equal(ax.collections[0].get_facecolor()[0],
+                                    (0, 0, 1, 1))
+
+        ax = df.plot.scatter(x='a', y='b', color='white')
+        tm.assert_numpy_array_equal(ax.collections[0].get_facecolor()[0],
+                                    (1, 1, 1, 1))
+
     @slow
     def test_plot_bar(self):
         df = DataFrame(randn(6, 4),

--- a/pandas/tests/test_graphics.py
+++ b/pandas/tests/test_graphics.py
@@ -800,8 +800,8 @@ class TestSeriesPlots(TestPlotBase):
         _check_plot_works(self.ts.hist)
         _check_plot_works(self.ts.hist, grid=False)
         _check_plot_works(self.ts.hist, figsize=(8, 10))
-        _check_plot_works(self.ts.hist, by=self.ts.index.month)
-        _check_plot_works(self.ts.hist, by=self.ts.index.month, bins=5)
+        _check_plot_works(self.ts.hist, filterwarnings='ignore', by=self.ts.index.month)
+        _check_plot_works(self.ts.hist, filterwarnings='ignore', by=self.ts.index.month, bins=5)
 
         fig, ax = self.plt.subplots(1, 1)
         _check_plot_works(self.ts.hist, ax=ax)
@@ -835,25 +835,32 @@ class TestSeriesPlots(TestPlotBase):
     def test_hist_layout_with_by(self):
         df = self.hist_df
 
-        axes = _check_plot_works(df.height.hist, by=df.gender, layout=(2, 1))
+        axes = _check_plot_works(df.height.hist, filterwarnings='ignore',
+                                 by=df.gender, layout=(2, 1))
         self._check_axes_shape(axes, axes_num=2, layout=(2, 1))
 
-        axes = _check_plot_works(df.height.hist, by=df.gender, layout=(3, -1))
+        axes = _check_plot_works(df.height.hist, filterwarnings='ignore',
+                                 by=df.gender, layout=(3, -1))
         self._check_axes_shape(axes, axes_num=2, layout=(3, 1))
 
-        axes = _check_plot_works(df.height.hist, by=df.category, layout=(4, 1))
+        axes = _check_plot_works(df.height.hist, filterwarnings='ignore',
+                                 by=df.category, layout=(4, 1))
         self._check_axes_shape(axes, axes_num=4, layout=(4, 1))
 
-        axes = _check_plot_works(df.height.hist, by=df.category, layout=(2, -1))
+        axes = _check_plot_works(df.height.hist, filterwarnings='ignore',
+                                 by=df.category, layout=(2, -1))
         self._check_axes_shape(axes, axes_num=4, layout=(2, 2))
 
-        axes = _check_plot_works(df.height.hist, by=df.category, layout=(3, -1))
+        axes = _check_plot_works(df.height.hist, filterwarnings='ignore',
+                                 by=df.category, layout=(3, -1))
         self._check_axes_shape(axes, axes_num=4, layout=(3, 2))
 
-        axes = _check_plot_works(df.height.hist, by=df.category, layout=(-1, 4))
+        axes = _check_plot_works(df.height.hist, filterwarnings='ignore',
+                                 by=df.category, layout=(-1, 4))
         self._check_axes_shape(axes, axes_num=4, layout=(1, 4))
 
-        axes = _check_plot_works(df.height.hist, by=df.classroom, layout=(2, 2))
+        axes = _check_plot_works(df.height.hist, filterwarnings='ignore',
+                                 by=df.classroom, layout=(2, 2))
         self._check_axes_shape(axes, axes_num=3, layout=(2, 2))
 
         axes = df.height.hist(by=df.category, layout=(4, 2), figsize=(12, 7))
@@ -1247,14 +1254,16 @@ class TestDataFramePlots(TestPlotBase):
     @slow
     def test_plot(self):
         df = self.tdf
-        _check_plot_works(df.plot, grid=False)
-        axes = _check_plot_works(df.plot, subplots=True)
+        _check_plot_works(df.plot, filterwarnings='ignore', grid=False)
+        axes = _check_plot_works(df.plot, filterwarnings='ignore', subplots=True)
         self._check_axes_shape(axes, axes_num=4, layout=(4, 1))
 
-        axes = _check_plot_works(df.plot, subplots=True, layout=(-1, 2))
+        axes = _check_plot_works(df.plot, filterwarnings='ignore',
+                                 subplots=True, layout=(-1, 2))
         self._check_axes_shape(axes, axes_num=4, layout=(2, 2))
 
-        axes = _check_plot_works(df.plot, subplots=True, use_index=False)
+        axes = _check_plot_works(df.plot, filterwarnings='ignore',
+                                 subplots=True, use_index=False)
         self._check_axes_shape(axes, axes_num=4, layout=(4, 1))
 
         df = DataFrame({'x': [1, 2], 'y': [3, 4]})
@@ -1263,13 +1272,14 @@ class TestDataFramePlots(TestPlotBase):
 
         df = DataFrame(np.random.rand(10, 3),
                        index=list(string.ascii_letters[:10]))
+
         _check_plot_works(df.plot, use_index=True)
         _check_plot_works(df.plot, sort_columns=False)
         _check_plot_works(df.plot, yticks=[1, 5, 10])
         _check_plot_works(df.plot, xticks=[1, 5, 10])
         _check_plot_works(df.plot, ylim=(-100, 100), xlim=(-100, 100))
 
-        _check_plot_works(df.plot, subplots=True, title='blah')
+        _check_plot_works(df.plot, filterwarnings='ignore', subplots=True, title='blah')
         # We have to redo it here because _check_plot_works does two plots, once without an ax
         # kwarg and once with an ax kwarg and the new sharex behaviour does not remove the
         # visibility of the latter axis (as ax is present).
@@ -2083,7 +2093,7 @@ class TestDataFramePlots(TestPlotBase):
 
         _check_plot_works(df.plot.bar)
         _check_plot_works(df.plot.bar, legend=False)
-        _check_plot_works(df.plot.bar, subplots=True)
+        _check_plot_works(df.plot.bar, filterwarnings='ignore', subplots=True)
         _check_plot_works(df.plot.bar, stacked=True)
 
         df = DataFrame(randn(10, 15),
@@ -2300,7 +2310,7 @@ class TestDataFramePlots(TestPlotBase):
         self._check_text_labels(ax.get_yticklabels(), labels)
         self.assertEqual(len(ax.lines), self.bp_n_objects * len(numeric_cols))
 
-        axes = _check_plot_works(df.plot.box, subplots=True,
+        axes = _check_plot_works(df.plot.box, filterwarnings='ignore', subplots=True,
                                  vert=False, logx=True)
         self._check_axes_shape(axes, axes_num=3, layout=(1, 3))
         self._check_ax_scales(axes, xaxis='log')
@@ -2360,7 +2370,7 @@ class TestDataFramePlots(TestPlotBase):
         ax = df.plot(kind='kde', rot=20, fontsize=5)
         self._check_ticks_props(ax, xrot=20, xlabelsize=5, ylabelsize=5)
 
-        axes = _check_plot_works(df.plot, kind='kde', subplots=True)
+        axes = _check_plot_works(df.plot, filterwarnings='ignore', kind='kde', subplots=True)
         self._check_axes_shape(axes, axes_num=4, layout=(4, 1))
 
         axes = df.plot(kind='kde', logy=True, subplots=True)
@@ -2387,7 +2397,7 @@ class TestDataFramePlots(TestPlotBase):
         expected = [com.pprint_thing(c) for c in df.columns]
         self._check_legend_labels(ax, labels=expected)
 
-        axes = _check_plot_works(df.plot.hist, subplots=True, logy=True)
+        axes = _check_plot_works(df.plot.hist, filterwarnings='ignore', subplots=True, logy=True)
         self._check_axes_shape(axes, axes_num=4, layout=(4, 1))
         self._check_ax_scales(axes, yaxis='log')
 
@@ -3093,7 +3103,7 @@ class TestDataFramePlots(TestPlotBase):
         ax = _check_plot_works(df.plot.pie, y=2)
         self._check_text_labels(ax.texts, df.index)
 
-        axes = _check_plot_works(df.plot.pie, subplots=True)
+        axes = _check_plot_works(df.plot.pie, filterwarnings='ignore', subplots=True)
         self.assertEqual(len(axes), len(df.columns))
         for ax in axes:
             self._check_text_labels(ax.texts, df.index)
@@ -3102,7 +3112,7 @@ class TestDataFramePlots(TestPlotBase):
 
         labels = ['A', 'B', 'C', 'D', 'E']
         color_args = ['r', 'g', 'b', 'c', 'm']
-        axes = _check_plot_works(df.plot.pie, subplots=True,
+        axes = _check_plot_works(df.plot.pie, filterwarnings='ignore', subplots=True,
                                  labels=labels, colors=color_args)
         self.assertEqual(len(axes), len(df.columns))
 
@@ -3156,7 +3166,8 @@ class TestDataFramePlots(TestPlotBase):
             self._check_has_errorbars(ax, xerr=2, yerr=2)
             ax = _check_plot_works(df.plot, xerr=0.2, yerr=0.2, kind=kind)
             self._check_has_errorbars(ax, xerr=2, yerr=2)
-            axes = _check_plot_works(df.plot, yerr=df_err, xerr=df_err, subplots=True, kind=kind)
+            axes = _check_plot_works(df.plot, filterwarnings='ignore', yerr=df_err,
+                                     xerr=df_err, subplots=True, kind=kind)
             self._check_has_errorbars(axes, xerr=1, yerr=1)
 
         ax = _check_plot_works((df+1).plot, yerr=df_err, xerr=df_err, kind='bar', log=True)
@@ -3245,7 +3256,8 @@ class TestDataFramePlots(TestPlotBase):
             self._check_has_errorbars(ax, xerr=0, yerr=1)
             ax = _check_plot_works(tdf.plot, yerr=tdf_err, kind=kind)
             self._check_has_errorbars(ax, xerr=0, yerr=2)
-            axes = _check_plot_works(tdf.plot, kind=kind, yerr=tdf_err, subplots=True)
+            axes = _check_plot_works(tdf.plot, filterwarnings='ignore', kind=kind,
+                                     yerr=tdf_err, subplots=True)
             self._check_has_errorbars(axes, xerr=0, yerr=1)
 
     def test_errorbar_asymmetrical(self):
@@ -3690,37 +3702,38 @@ def assert_is_valid_plot_return_object(objs):
                  ''.format(objs.__class__.__name__))
 
 
-def _check_plot_works(f, *args, **kwargs):
+def _check_plot_works(f, filterwarnings='always', **kwargs):
     import matplotlib.pyplot as plt
     ret = None
-
-    try:
+    with warnings.catch_warnings():
+        warnings.simplefilter(filterwarnings)
         try:
-            fig = kwargs['figure']
-        except KeyError:
-            fig = plt.gcf()
+            try:
+                fig = kwargs['figure']
+            except KeyError:
+                fig = plt.gcf()
 
-        plt.clf()
+            plt.clf()
 
-        ax = kwargs.get('ax', fig.add_subplot(211))
-        ret = f(*args, **kwargs)
+            ax = kwargs.get('ax', fig.add_subplot(211))
+            ret = f(**kwargs)
 
-        assert_is_valid_plot_return_object(ret)
-
-        try:
-            kwargs['ax'] = fig.add_subplot(212)
-            ret = f(*args, **kwargs)
-        except Exception:
-            pass
-        else:
             assert_is_valid_plot_return_object(ret)
 
-        with ensure_clean(return_filelike=True) as path:
-            plt.savefig(path)
-    finally:
-        tm.close(fig)
+            try:
+                kwargs['ax'] = fig.add_subplot(212)
+                ret = f(**kwargs)
+            except Exception:
+                pass
+            else:
+                assert_is_valid_plot_return_object(ret)
 
-    return ret
+            with ensure_clean(return_filelike=True) as path:
+                plt.savefig(path)
+        finally:
+            tm.close(fig)
+
+        return ret
 
 def _generate_4_axes_via_gridspec():
     import matplotlib.pyplot as plt

--- a/pandas/tests/test_graphics.py
+++ b/pandas/tests/test_graphics.py
@@ -74,13 +74,22 @@ class TestPlotBase(tm.TestCase):
                                       'weight': random.normal(161, 32, size=n),
                                       'category': random.randint(4, size=n)})
 
-        if str(mpl.__version__) >= LooseVersion('1.4'):
+        self.mpl_le_1_2_1 = plotting._mpl_le_1_2_1()
+        self.mpl_ge_1_3_1 = plotting._mpl_ge_1_3_1()
+        self.mpl_ge_1_4_0 = plotting._mpl_ge_1_4_0()
+        self.mpl_ge_1_5_0 = plotting._mpl_ge_1_5_0()
+
+        if self.mpl_ge_1_4_0:
             self.bp_n_objects = 7
         else:
             self.bp_n_objects = 8
+        if self.mpl_ge_1_5_0:
+            # 1.5 added PolyCollections to legend handler
+            # so we have twice as many items.
+            self.polycollection_factor = 2
+        else:
+            self.polycollection_factor = 1
 
-        self.mpl_le_1_2_1 = str(mpl.__version__) <= LooseVersion('1.2.1')
-        self.mpl_ge_1_3_1 = str(mpl.__version__) >= LooseVersion('1.3.1')
 
     def tearDown(self):
         tm.close()
@@ -183,7 +192,7 @@ class TestPlotBase(tm.TestCase):
         """
 
         from matplotlib.lines import Line2D
-        from matplotlib.collections import Collection
+        from matplotlib.collections import Collection, PolyCollection
         conv = self.colorconverter
         if linecolors is not None:
 
@@ -197,6 +206,8 @@ class TestPlotBase(tm.TestCase):
                     result = patch.get_color()
                     # Line2D may contains string color expression
                     result = conv.to_rgba(result)
+                elif isinstance(patch, PolyCollection):
+                    result = tuple(patch.get_edgecolor()[0])
                 else:
                     result = patch.get_edgecolor()
 
@@ -472,6 +483,21 @@ class TestPlotBase(tm.TestCase):
                 obj.plot(kind=kind, grid=True, **kws)
                 self.assertTrue(is_grid_on())
 
+    def _maybe_unpack_cycler(self, rcParams, field='color'):
+        """
+        Compat layer for MPL 1.5 change to color cycle
+
+        Before: plt.rcParams['axes.color_cycle'] -> ['b', 'g', 'r'...]
+        After : plt.rcParams['axes.prop_cycle'] -> cycler(...)
+        """
+        if self.mpl_ge_1_5_0:
+            cyl = rcParams['axes.prop_cycle']
+            colors = [v[field] for v in cyl]
+        else:
+            colors = rcParams['axes.color_cycle']
+        return colors
+
+
 @tm.mplskip
 class TestSeriesPlots(TestPlotBase):
 
@@ -536,9 +562,13 @@ class TestSeriesPlots(TestPlotBase):
 
     def test_dont_modify_rcParams(self):
         # GH 8242
-        colors = self.plt.rcParams['axes.color_cycle']
+        if self.mpl_ge_1_5_0:
+            key = 'axes.prop_cycle'
+        else:
+            key = 'axes.color_cycle'
+        colors = self.plt.rcParams[key]
         Series([1, 2, 3]).plot()
-        self.assertEqual(colors, self.plt.rcParams['axes.color_cycle'])
+        self.assertEqual(colors, self.plt.rcParams[key])
 
     def test_ts_line_lim(self):
         ax = self.ts.plot()
@@ -1109,7 +1139,8 @@ class TestSeriesPlots(TestPlotBase):
             s.plot(yerr=np.arange(11))
 
         s_err = ['zzz']*10
-        with tm.assertRaises(TypeError):
+        # in mpl 1.5+ this is a TypeError
+        with tm.assertRaises((ValueError, TypeError)):
             s.plot(yerr=s_err)
 
     def test_table(self):
@@ -1183,7 +1214,10 @@ class TestSeriesPlots(TestPlotBase):
     def test_time_series_plot_color_with_empty_kwargs(self):
         import matplotlib as mpl
 
-        def_colors = mpl.rcParams['axes.color_cycle']
+        if self.mpl_ge_1_5_0:
+            def_colors = self._maybe_unpack_cycler(mpl.rcParams)
+        else:
+            def_colors = mpl.rcParams['axes.color_cycle']
         index = date_range('1/1/2000', periods=12)
         s = Series(np.arange(1, 13), index=index)
 
@@ -1292,7 +1326,11 @@ class TestDataFramePlots(TestPlotBase):
         fig, ax = self.plt.subplots()
         axes = df.plot.bar(subplots=True, ax=ax)
         self.assertEqual(len(axes), 1)
-        self.assertIs(ax.get_axes(), axes[0])
+        if self.mpl_ge_1_5_0:
+            result = ax.axes
+        else:
+            result = ax.get_axes()  # deprecated
+        self.assertIs(result, axes[0])
 
     def test_color_and_style_arguments(self):
         df = DataFrame({'x': [1, 2], 'y': [3, 4]})
@@ -1802,8 +1840,7 @@ class TestDataFramePlots(TestPlotBase):
     @slow
     def test_bar_colors(self):
         import matplotlib.pyplot as plt
-
-        default_colors = plt.rcParams.get('axes.color_cycle')
+        default_colors = self._maybe_unpack_cycler(plt.rcParams)
 
         df = DataFrame(randn(5, 5))
         ax = df.plot.bar()
@@ -2326,6 +2363,7 @@ class TestDataFramePlots(TestPlotBase):
 
     @slow
     def test_hist_df(self):
+        from matplotlib.patches import Rectangle
         if self.mpl_le_1_2_1:
             raise nose.SkipTest("not supported in matplotlib <= 1.2.x")
 
@@ -2346,11 +2384,14 @@ class TestDataFramePlots(TestPlotBase):
 
         ax = series.plot.hist(normed=True, cumulative=True, bins=4)
         # height of last bin (index 5) must be 1.0
-        self.assertAlmostEqual(ax.get_children()[5].get_height(), 1.0)
+        rects = [x for x in ax.get_children() if isinstance(x, Rectangle)]
+        self.assertAlmostEqual(rects[-1].get_height(), 1.0)
         tm.close()
 
         ax = series.plot.hist(cumulative=True, bins=4)
-        self.assertAlmostEqual(ax.get_children()[5].get_height(), 100.0)
+        rects = [x for x in ax.get_children() if isinstance(x, Rectangle)]
+
+        self.assertAlmostEqual(rects[-2].get_height(), 100.0)
         tm.close()
 
         # if horizontal, yticklabels are rotated
@@ -2626,7 +2667,7 @@ class TestDataFramePlots(TestPlotBase):
     def test_line_colors_and_styles_subplots(self):
         # GH 9894
         from matplotlib import cm
-        default_colors = self.plt.rcParams.get('axes.color_cycle')
+        default_colors = self._maybe_unpack_cycler(self.plt.rcParams)
 
         df = DataFrame(randn(5, 5))
 
@@ -2698,7 +2739,8 @@ class TestDataFramePlots(TestPlotBase):
 
         handles, labels = ax.get_legend_handles_labels()
         # legend is stored as Line2D, thus check linecolors
-        self._check_colors(handles, linecolors=custom_colors)
+        linehandles = [x for x in handles if not isinstance(x, PolyCollection)]
+        self._check_colors(linehandles, linecolors=custom_colors)
         for h in handles:
             self.assertTrue(h.get_alpha() is None)
         tm.close()
@@ -2710,12 +2752,13 @@ class TestDataFramePlots(TestPlotBase):
         self._check_colors(poly, facecolors=jet_colors)
 
         handles, labels = ax.get_legend_handles_labels()
-        self._check_colors(handles, linecolors=jet_colors)
+        linehandles = [x for x in handles if not isinstance(x, PolyCollection)]
+        self._check_colors(linehandles, linecolors=jet_colors)
         for h in handles:
             self.assertTrue(h.get_alpha() is None)
         tm.close()
 
-        # When stacked=True, alpha is set to 0.5
+        # When stacked=False, alpha is set to 0.5
         ax = df.plot.area(colormap=cm.jet, stacked=False)
         self._check_colors(ax.get_lines(), linecolors=jet_colors)
         poly = [o for o in ax.get_children() if isinstance(o, PolyCollection)]
@@ -2724,13 +2767,13 @@ class TestDataFramePlots(TestPlotBase):
 
         handles, labels = ax.get_legend_handles_labels()
         # Line2D can't have alpha in its linecolor
-        self._check_colors(handles, linecolors=jet_colors)
+        self._check_colors(handles[:len(jet_colors)], linecolors=jet_colors)
         for h in handles:
             self.assertEqual(h.get_alpha(), 0.5)
 
     @slow
     def test_hist_colors(self):
-        default_colors = self.plt.rcParams.get('axes.color_cycle')
+        default_colors = self._maybe_unpack_cycler(self.plt.rcParams)
 
         df = DataFrame(randn(5, 5))
         ax = df.plot.hist()
@@ -2791,7 +2834,7 @@ class TestDataFramePlots(TestPlotBase):
         _skip_if_no_scipy_gaussian_kde()
 
         from matplotlib import cm
-        default_colors = self.plt.rcParams.get('axes.color_cycle')
+        default_colors = self._maybe_unpack_cycler(self.plt.rcParams)
 
         df = DataFrame(randn(5, 5))
 
@@ -2853,7 +2896,7 @@ class TestDataFramePlots(TestPlotBase):
             self._check_colors(bp['fliers'], linecolors=[fliers_c] * len(bp['fliers']))
             self._check_colors(bp['caps'], linecolors=[caps_c] * len(bp['caps']))
 
-        default_colors = self.plt.rcParams.get('axes.color_cycle')
+        default_colors = self._maybe_unpack_cycler(self.plt.rcParams)
 
         df = DataFrame(randn(5, 5))
         bp = df.plot.box(return_type='dict')
@@ -2900,12 +2943,17 @@ class TestDataFramePlots(TestPlotBase):
 
     def test_default_color_cycle(self):
         import matplotlib.pyplot as plt
-        plt.rcParams['axes.color_cycle'] = list('rgbk')
+        colors = list('rgbk')
+        if self.mpl_ge_1_5_0:
+            import cycler
+            plt.rcParams['axes.prop_cycle'] = cycler.cycler('color', colors)
+        else:
+            plt.rcParams['axes.color_cycle'] = colors
 
         df = DataFrame(randn(5, 3))
         ax = df.plot()
 
-        expected = plt.rcParams['axes.color_cycle'][:3]
+        expected = self._maybe_unpack_cycler(plt.rcParams)[:3]
         self._check_colors(ax.get_lines(), linecolors=expected)
 
     def test_unordered_ts(self):
@@ -3125,7 +3173,7 @@ class TestDataFramePlots(TestPlotBase):
             df.plot(yerr=np.random.randn(11))
 
         df_err = DataFrame({'x': ['zzz']*12, 'y': ['zzz']*12})
-        with tm.assertRaises(TypeError):
+        with tm.assertRaises((ValueError, TypeError)):
             df.plot(yerr=df_err)
 
     @slow

--- a/pandas/tests/test_graphics_others.py
+++ b/pandas/tests/test_graphics_others.py
@@ -607,6 +607,8 @@ class TestDataFrameGroupByPlots(TestPlotBase):
 
     @slow
     def test_grouped_hist_legacy(self):
+        from matplotlib.patches import Rectangle
+
         df = DataFrame(randn(500, 2), columns=['A', 'B'])
         df['C'] = np.random.randint(0, 4, 500)
         df['D'] = ['X'] * 500
@@ -633,7 +635,8 @@ class TestDataFrameGroupByPlots(TestPlotBase):
                                      xlabelsize=xf, xrot=xrot, ylabelsize=yf, yrot=yrot)
         # height of last bin (index 5) must be 1.0
         for ax in axes.ravel():
-            height = ax.get_children()[5].get_height()
+            rects = [x for x in ax.get_children() if isinstance(x, Rectangle)]
+            height = rects[-1].get_height()
             self.assertAlmostEqual(height, 1.0)
         self._check_ticks_props(axes, xlabelsize=xf, xrot=xrot,
                                 ylabelsize=yf, yrot=yrot)

--- a/pandas/tests/test_graphics_others.py
+++ b/pandas/tests/test_graphics_others.py
@@ -161,8 +161,8 @@ class TestSeriesPlots(TestPlotBase):
     @slow
     def test_autocorrelation_plot(self):
         from pandas.tools.plotting import autocorrelation_plot
-        _check_plot_works(autocorrelation_plot, self.ts)
-        _check_plot_works(autocorrelation_plot, self.ts.values)
+        _check_plot_works(autocorrelation_plot, series=self.ts)
+        _check_plot_works(autocorrelation_plot, series=self.ts.values)
 
         ax = autocorrelation_plot(self.ts, label='Test')
         self._check_legend_labels(ax, labels=['Test'])
@@ -170,13 +170,13 @@ class TestSeriesPlots(TestPlotBase):
     @slow
     def test_lag_plot(self):
         from pandas.tools.plotting import lag_plot
-        _check_plot_works(lag_plot, self.ts)
-        _check_plot_works(lag_plot, self.ts, lag=5)
+        _check_plot_works(lag_plot, series=self.ts)
+        _check_plot_works(lag_plot, series=self.ts, lag=5)
 
     @slow
     def test_bootstrap_plot(self):
         from pandas.tools.plotting import bootstrap_plot
-        _check_plot_works(bootstrap_plot, self.ts, size=10)
+        _check_plot_works(bootstrap_plot, series=self.ts, size=10)
 
 
 @tm.mplskip
@@ -210,7 +210,7 @@ class TestDataFramePlots(TestPlotBase):
         _check_plot_works(df.boxplot, column='one', by=['indic', 'indic2'])
         _check_plot_works(df.boxplot, by='indic')
         _check_plot_works(df.boxplot, by=['indic', 'indic2'])
-        _check_plot_works(plotting.boxplot, df['one'], return_type='dict')
+        _check_plot_works(plotting.boxplot, data=df['one'], return_type='dict')
         _check_plot_works(df.boxplot, notch=1, return_type='dict')
         _check_plot_works(df.boxplot, by='indic', notch=1)
 
@@ -304,6 +304,7 @@ class TestDataFramePlots(TestPlotBase):
 
     @slow
     def test_hist_df_legacy(self):
+        from matplotlib.patches import Rectangle
         _check_plot_works(self.hist_df.hist)
 
         # make sure layout is handled
@@ -347,7 +348,8 @@ class TestDataFramePlots(TestPlotBase):
         # make sure kwargs to hist are handled
         ax = ser.hist(normed=True, cumulative=True, bins=4)
         # height of last bin (index 5) must be 1.0
-        self.assertAlmostEqual(ax.get_children()[5].get_height(), 1.0)
+        rects = [x for x in ax.get_children() if isinstance(x, Rectangle)]
+        self.assertAlmostEqual(rects[-1].get_height(), 1.0)
 
         tm.close()
         ax = ser.hist(log=True)
@@ -413,9 +415,9 @@ class TestDataFramePlots(TestPlotBase):
         def scat2(x, y, by=None, ax=None, figsize=None):
             return plotting.scatter_plot(df, x, y, by, ax, figsize=None)
 
-        _check_plot_works(scat2, 0, 1)
+        _check_plot_works(scat2, x=0, y=1)
         grouper = Series(np.repeat([1, 2, 3, 4, 5], 20), df.index)
-        _check_plot_works(scat2, 0, 1, by=grouper)
+        _check_plot_works(scat2, x=0, y=1, by=grouper)
 
     def test_scatter_matrix_axis(self):
         tm._skip_if_no_scipy()
@@ -424,7 +426,8 @@ class TestDataFramePlots(TestPlotBase):
         with tm.RNGContext(42):
             df = DataFrame(randn(100, 3))
 
-        axes = _check_plot_works(scatter_matrix, df, range_padding=.1)
+        axes = _check_plot_works(scatter_matrix, filterwarnings='always', frame=df,
+                                 range_padding=.1)
         axes0_labels = axes[0][0].yaxis.get_majorticklabels()
         # GH 5662
         expected = ['-2', '-1', '0', '1', '2']
@@ -432,7 +435,8 @@ class TestDataFramePlots(TestPlotBase):
         self._check_ticks_props(axes, xlabelsize=8, xrot=90, ylabelsize=8, yrot=0)
 
         df[0] = ((df[0] - 2) / 3)
-        axes = _check_plot_works(scatter_matrix, df, range_padding=.1)
+        axes = _check_plot_works(scatter_matrix, filterwarnings='always', frame=df,
+                                 range_padding=.1)
         axes0_labels = axes[0][0].yaxis.get_majorticklabels()
         expected = ['-1.2', '-1.0', '-0.8', '-0.6', '-0.4', '-0.2', '0.0']
         self._check_text_labels(axes0_labels, expected)
@@ -445,17 +449,17 @@ class TestDataFramePlots(TestPlotBase):
 
         df = self.iris
 
-        _check_plot_works(andrews_curves, df, 'Name')
+        _check_plot_works(andrews_curves, frame=df, class_column='Name')
 
         rgba = ('#556270', '#4ECDC4', '#C7F464')
-        ax = _check_plot_works(andrews_curves, df, 'Name', color=rgba)
+        ax = _check_plot_works(andrews_curves, frame=df, class_column='Name', color=rgba)
         self._check_colors(ax.get_lines()[:10], linecolors=rgba, mapping=df['Name'][:10])
 
         cnames = ['dodgerblue', 'aquamarine', 'seagreen']
-        ax = _check_plot_works(andrews_curves, df, 'Name', color=cnames)
+        ax = _check_plot_works(andrews_curves, frame=df, class_column='Name', color=cnames)
         self._check_colors(ax.get_lines()[:10], linecolors=cnames, mapping=df['Name'][:10])
 
-        ax = _check_plot_works(andrews_curves, df, 'Name', colormap=cm.jet)
+        ax = _check_plot_works(andrews_curves, frame=df, class_column='Name', colormap=cm.jet)
         cmaps = lmap(cm.jet, np.linspace(0, 1, df['Name'].nunique()))
         self._check_colors(ax.get_lines()[:10], linecolors=cmaps, mapping=df['Name'][:10])
 
@@ -478,23 +482,23 @@ class TestDataFramePlots(TestPlotBase):
 
         df = self.iris
 
-        ax = _check_plot_works(parallel_coordinates, df, 'Name')
+        ax = _check_plot_works(parallel_coordinates, frame=df, class_column='Name')
         nlines = len(ax.get_lines())
         nxticks = len(ax.xaxis.get_ticklabels())
 
         rgba = ('#556270', '#4ECDC4', '#C7F464')
-        ax = _check_plot_works(parallel_coordinates, df, 'Name', color=rgba)
+        ax = _check_plot_works(parallel_coordinates, frame=df, class_column='Name', color=rgba)
         self._check_colors(ax.get_lines()[:10], linecolors=rgba, mapping=df['Name'][:10])
 
         cnames = ['dodgerblue', 'aquamarine', 'seagreen']
-        ax = _check_plot_works(parallel_coordinates, df, 'Name', color=cnames)
+        ax = _check_plot_works(parallel_coordinates, frame=df, class_column='Name', color=cnames)
         self._check_colors(ax.get_lines()[:10], linecolors=cnames, mapping=df['Name'][:10])
 
-        ax = _check_plot_works(parallel_coordinates, df, 'Name', colormap=cm.jet)
+        ax = _check_plot_works(parallel_coordinates, frame=df, class_column='Name', colormap=cm.jet)
         cmaps = lmap(cm.jet, np.linspace(0, 1, df['Name'].nunique()))
         self._check_colors(ax.get_lines()[:10], linecolors=cmaps, mapping=df['Name'][:10])
 
-        ax = _check_plot_works(parallel_coordinates, df, 'Name', axvlines=False)
+        ax = _check_plot_works(parallel_coordinates, frame=df, class_column='Name', axvlines=False)
         assert len(ax.get_lines()) == (nlines - nxticks)
 
         colors = ['b', 'g', 'r']
@@ -517,20 +521,20 @@ class TestDataFramePlots(TestPlotBase):
         from matplotlib import cm
 
         df = self.iris
-        _check_plot_works(radviz, df, 'Name')
+        _check_plot_works(radviz, frame=df, class_column='Name')
 
         rgba = ('#556270', '#4ECDC4', '#C7F464')
-        ax = _check_plot_works(radviz, df, 'Name', color=rgba)
+        ax = _check_plot_works(radviz, frame=df, class_column='Name', color=rgba)
         # skip Circle drawn as ticks
         patches = [p for p in ax.patches[:20] if p.get_label() != '']
         self._check_colors(patches[:10], facecolors=rgba, mapping=df['Name'][:10])
 
         cnames = ['dodgerblue', 'aquamarine', 'seagreen']
-        _check_plot_works(radviz, df, 'Name', color=cnames)
+        _check_plot_works(radviz, frame=df, class_column='Name', color=cnames)
         patches = [p for p in ax.patches[:20] if p.get_label() != '']
         self._check_colors(patches, facecolors=cnames, mapping=df['Name'][:10])
 
-        _check_plot_works(radviz, df, 'Name', colormap=cm.jet)
+        _check_plot_works(radviz, frame=df, class_column='Name', colormap=cm.jet)
         cmaps = lmap(cm.jet, np.linspace(0, 1, df['Name'].nunique()))
         patches = [p for p in ax.patches[:20] if p.get_label() != '']
         self._check_colors(patches, facecolors=cmaps, mapping=df['Name'][:10])

--- a/pandas/tools/plotting.py
+++ b/pandas/tools/plotting.py
@@ -24,12 +24,12 @@ from pandas.tseries.offsets import DateOffset
 from pandas.compat import range, lrange, lmap, map, zip, string_types
 import pandas.compat as compat
 from pandas.util.decorators import Appender
-
 try:  # mpl optional
     import pandas.tseries.converter as conv
     conv.register()  # needs to override so set_xlim works with str/number
 except ImportError:
     pass
+
 
 # Extracted from https://gist.github.com/huyng/816622
 # this is the rcParams set when setting display.with_mpl_style
@@ -96,6 +96,48 @@ mpl_stylesheet = {
      'ytick.minor.pad': 6.0,
      'ytick.minor.size': 0.0
 }
+
+
+def _mpl_le_1_2_1():
+    try:
+        import matplotlib as mpl
+        return (str(mpl.__version__) <= LooseVersion('1.2.1') and
+                str(mpl.__version__)[0] != '0')
+    except ImportError:
+        return False
+
+def _mpl_ge_1_3_1():
+    try:
+        import matplotlib
+        # The or v[0] == '0' is because their versioneer is
+        # messed up on dev
+        return (matplotlib.__version__ >= LooseVersion('1.3.1')
+                or matplotlib.__version__[0] == '0')
+    except ImportError:
+        return False
+
+def _mpl_ge_1_4_0():
+    try:
+        import matplotlib
+        return (matplotlib.__version__  >= LooseVersion('1.4')
+                or matplotlib.__version__[0] == '0')
+    except ImportError:
+        return False
+
+def _mpl_ge_1_5_0():
+    try:
+        import matplotlib
+        return (matplotlib.__version__  >= LooseVersion('1.5')
+                or matplotlib.__version__[0] == '0')
+    except ImportError:
+        return False
+
+if _mpl_ge_1_5_0():
+    # Compat with mp 1.5, which uses cycler.
+    import cycler
+    colors = mpl_stylesheet.pop('axes.color_cycle')
+    mpl_stylesheet['axes.prop_cycle'] = cycler.cycler('color_cycle', colors)
+
 
 def _get_standard_kind(kind):
     return {'density': 'kde'}.get(kind, kind)
@@ -784,7 +826,6 @@ class MPLPlot(object):
     _layout_type = 'vertical'
     _default_rot = 0
     orientation = None
-
     _pop_attributes = ['label', 'style', 'logy', 'logx', 'loglog',
                        'mark_right', 'stacked']
     _attr_defaults = {'logy': False, 'logx': False, 'loglog': False,
@@ -973,8 +1014,9 @@ class MPLPlot(object):
         else:
             # otherwise, create twin axes
             orig_ax, new_ax = ax, ax.twinx()
-            new_ax._get_lines.color_cycle = orig_ax._get_lines.color_cycle
-
+            # TODO: use Matplotlib public API when available
+            new_ax._get_lines = orig_ax._get_lines
+            new_ax._get_patches_for_fill = orig_ax._get_patches_for_fill
             orig_ax.right_ax, new_ax.left_ax = new_ax, orig_ax
 
             if not self._has_plotted_object(orig_ax):  # no data on left y
@@ -1196,6 +1238,14 @@ class MPLPlot(object):
     def plt(self):
         import matplotlib.pyplot as plt
         return plt
+
+    @staticmethod
+    def mpl_ge_1_3_1():
+        return _mpl_ge_1_3_1()
+
+    @staticmethod
+    def mpl_ge_1_5_0():
+        return _mpl_ge_1_5_0()
 
     _need_to_set_index = False
 
@@ -1474,9 +1524,6 @@ class ScatterPlot(PlanePlot):
         self.c = c
 
     def _make_plot(self):
-        import matplotlib as mpl
-        mpl_ge_1_3_1 = str(mpl.__version__) >= LooseVersion('1.3.1')
-
         x, y, c, data = self.x, self.y, self.c, self.data
         ax = self.axes[0]
 
@@ -1505,7 +1552,7 @@ class ScatterPlot(PlanePlot):
         if cb:
             img = ax.collections[0]
             kws = dict(ax=ax)
-            if mpl_ge_1_3_1:
+            if self.mpl_ge_1_3_1():
                 kws['label'] = c if c_is_column else ''
             self.fig.colorbar(img, **kws)
 
@@ -1636,6 +1683,11 @@ class LinePlot(MPLPlot):
 
         # Set ax with freq info
         _decorate_axes(ax, freq, kwds)
+        # digging deeper
+        if hasattr(ax, 'left_ax'):
+            _decorate_axes(ax.left_ax, freq, kwds)
+        if hasattr(ax, 'right_ax'):
+            _decorate_axes(ax.right_ax, freq, kwds)
         ax._plot_data.append((data, cls._kind, kwds))
 
         lines = cls._plot(ax, data.index, data.values, style=style, **kwds)
@@ -1743,6 +1795,8 @@ class AreaPlot(LinePlot):
         if not 'color' in kwds:
             kwds['color'] = lines[0].get_color()
 
+        if cls.mpl_ge_1_5_0(): # mpl 1.5 added real support for poly legends
+            kwds.pop('label')
         ax.fill_between(xdata, start, y_values, **kwds)
         cls._update_stacker(ax, stacking_id, y)
         return lines

--- a/pandas/tools/plotting.py
+++ b/pandas/tools/plotting.py
@@ -1535,9 +1535,13 @@ class ScatterPlot(PlanePlot):
         # pandas uses colormap, matplotlib uses cmap.
         cmap = self.colormap or 'Greys'
         cmap = self.plt.cm.get_cmap(cmap)
-
-        if c is None:
+        color = self.kwds.pop("color", None)
+        if c is not None and color is not None:
+            raise TypeError('Specify exactly one of `c` and `color`')
+        elif c is None and color is None:
             c_values = self.plt.rcParams['patch.facecolor']
+        elif color is not None:
+            c_values = color
         elif c_is_column:
             c_values = self.data[c].values
         else:

--- a/pandas/tseries/tests/test_plotting.py
+++ b/pandas/tseries/tests/test_plotting.py
@@ -483,6 +483,7 @@ class TestTSPlot(tm.TestCase):
         ts[5:25] = np.nan
         ax = ts.plot()
         lines = ax.get_lines()
+        tm._skip_if_mpl_1_5()
         self.assertEqual(len(lines), 1)
         l = lines[0]
         data = l.get_xydata()
@@ -532,6 +533,9 @@ class TestTSPlot(tm.TestCase):
         self.assertEqual(len(ax.right_ax.get_lines()), 1)
         l = lines[0]
         data = l.get_xydata()
+
+        tm._skip_if_mpl_1_5()
+
         tm.assertIsInstance(data, np.ma.core.MaskedArray)
         mask = data.mask
         self.assertTrue(mask[5:25, 1].all())

--- a/pandas/util/testing.py
+++ b/pandas/util/testing.py
@@ -196,6 +196,14 @@ def mplskip(cls):
     return cls
 
 
+def _skip_if_mpl_1_5():
+    import matplotlib
+    v = matplotlib.__version__
+    if v > LooseVersion('1.4.3') or v[0] == '0':
+        import nose
+        raise nose.SkipTest("matplotlib 1.5")
+
+
 def _skip_if_no_scipy():
     try:
         import scipy.stats

